### ISSUE TITLE
Support Hibernate 5.3.0 / JPA 2.2 (JDK8 Streams for pagination)

### ISSUE THREAD

**Issue by Blarne** [spring-projects/spring-boot#11725]
Based on functionality supporting streams JDK8 over JPA (automatic pagination) described in https://vladmihalcea.com/whats-new-in-jpa-2-2-stream-the-result-of-a-query-execution/ included JPA 2.2 with Hibernate 5.3.0.Beta1 - I recommend upgrade dependencies on Spring Boot Dependencies to:

```
<!-- https://mvnrepository.com/artifact/org.hibernate/hibernate-core -->
<dependency>
    <groupId>org.hibernate</groupId>
    <artifactId>hibernate-core</artifactId>
    <version>5.3.0.Beta1</version>
</dependency>

<!-- https://mvnrepository.com/artifact/javax.persistence/javax.persistence-api -->
<dependency>
    <groupId>javax.persistence</groupId>
    <artifactId>javax.persistence-api</artifactId>
    <version>2.2</version>
</dependency>
```
I can contribute on this issue, if possible.

**Comment by wilkinsona** [comment]
Thanks, but things aren't ready for this yet. We're very close to shipping Boot 2.0 and we cannot use a beta version of Hibernate in a Boot GA. We may also need to wait for Spring Framework and Spring Data JPA to support JPA 2.2 and Hibernate 5.3. Once they do, Boot's upgrade to Hibernate 5.3 will follow.

**Comment by Blarne** [comment]
Yes, undestand, I agree with stable version of HBM too.

**Comment by jhoeller** [comment]
FYI, we're tracking Hibernate ORM 5.3 support in [SPR-16303](https://jira.spring.io/browse/SPR-16303), aiming for the Spring Framework 5.1 timeframe in Q2 2018 (when Hibernate ORM 5.3 itself will hopefully be GA). For the time being, the 5.3 Beta 1 release is known to work fine with the upcoming Spring Framework 5.0.3, just with no dedicated support for any Hibernate 5.3 specific features there yet.

As for the JPA 2.2 API, that's explicitly supported in Spring Framework 5.0 already, including proper handling of result streams in non-transactional scenarios. Since JPA 2.2 doesn't make much difference for Spring's management at runtime, in particular in regular transactional setups, providers such as EclipseLink 2.7 or Hibernate ORM 5.3 should even work fine with Spring Framework 4.3.x as well.

**Comment by ask4gilles** [comment]
FYI, hibernate 5.3.0.Final has been released https://github.com/hibernate/hibernate-orm/releases/tag/5.3.0.Final

**Comment by wilkinsona** [comment]
Thanks, @ask4gilles. That brings us one step closer. Next on the list is https://jira.spring.io/browse/SPR-16569 which I am told is blocking Spring Data JPA from being Hibernate 5.3 compatible. The Data team are looking into it.

**Comment by heruan** [comment]
WildFly 13 has been released with ORM 5.3 provided in EE8 mode, it would be great to have also Spring updated now.

**Comment by heruan** [comment]
Now that [SPR-16569](https://jira.spring.io/browse/SPR-16569) has been resolved, is there anything else keeping 5.3 from being adopted by Spring Data?
